node/src/daser.rs

```diff
@@ -8,25 +8,27 @@
 //!
 //! [`Sample`]: celestia_types::sample::Sample
 
-use std::collections::HashSet;
+use std::collections::{HashSet, VecDeque};
 use std::sync::Arc;
 
+use celestia_tendermint::Time;
 use celestia_types::ExtendedHeader;
-use cid::Cid;
 use futures::stream::FuturesUnordered;
 use futures::StreamExt;
-use instant::Instant;
+use instant::{Duration, Instant};
 use rand::Rng;
 use tokio::select;
 use tokio_util::sync::CancellationToken;
-use tracing::{debug, error};
+use tracing::{debug, error, warn};
 
 use crate::executor::spawn;
-use crate::p2p::shwap::convert_cid;
-use crate::p2p::{P2p, P2pError};
-use crate::store::{Store, StoreError};
+use crate::p2p::shwap::sample_cid;
+use crate::p2p::{P2p, P2pError, GET_SAMPLE_TIMEOUT};
+use crate::store::{SamplingStatus, Store, StoreError};
+use crate::utils::FusedReusableBoxFuture;
 
 const MAX_SAMPLES_NEEDED: usize = 16;
+const SAMPLING_WINDOW: Duration = Duration::from_secs(14678036);
```

Comment:
If my calculation is correct, that's ~169.88 days. Where's that coming from? I thought sampling window was supposed to be 30 days.
Also could we do something like `Duration::from_secs($days * 24 * 60 * 60)`, so that it's more obvious what the value is?

Reply:
Good catch. It is a leftover from some manual tests I was doing.